.ref_replacement

```diff
@@ -0,0 +1,95 @@
+module type REF = sig
+  type 'a t
+
+  val ref_prefix : string
+  val make : unit -> 'a t
+  val null : 'a t
+  val string_of : 'a t -> string
+  val of_string : string -> 'a t
+
+  val make_dummy : string -> 'a t
+  val is_dummy : 'a t -> bool
+  val name_of_dummy : 'a t -> string
+  val pretty_string_of_dummy : 'a t -> string
+  val really_pretty_and_small : 'a t -> string
+
+  val rpc_of_t: 'b -> 'a t -> Rpc.t
+  val t_of_rpc: 'b -> Rpc.t -> 'a t
+end
+module Ref: REF = struct
+  (* BEGIN ref.ml from xen-api repository *)
```

Comment:
So we need to update this every time we update the file from which it is copied?

Reply:
Not really, we were copying only a function from that file, and it was an old version of it. I have updated that on my latest re-generation just to provide some additional prettyprinters. I have left the comments only to pinpoint where those functions came from.

Reply:
(plus that is not even the current ref.ml, I have removed the dependency from stdext and I have a PR ready to update the one in xapi with these changes)

Reply:
Once I am able to use `xapi-client` also that will be unneeded.